en/guide/connections.md

```diff
@@ -0,0 +1,84 @@
+
+The methods are used in the same way, and invoke a callback function with the UUID of the device that was discovered/disconnected. This UUID can then be used to get a `Device` object for managing the associated vehicle (see [Accessing Devices](#accessing-devices) below).
+
+The code fragment below shows how to register a callback (in this case the callback is a lambda function that just prints the UUID value to standard `cout`).
+
+```cpp
+DroneCore dc;
+... //add ports
+dc.register_on_discover([](uint64_t uuid) {
+    std::cout << "Discovered device with UUID: " << uuid << std::endl;
+});
+```
+
+## Iterating all Devices
+
+You can iterate all device UUIDs that have been detected by DroneCore 
+(since it was started, over all communication ports) using the [device_uuids()](../api_reference/classdronecore_1_1_drone_core.md#classdronecore_1_1_drone_core_1a20490717da3893be2c6965b905a7c1db) method. This returns a vector of [UUID](../api_reference/classdronecore_1_1_info.md#classdronecore_1_1_info_1a49c7dd5f1a369c8296f0c3a2443bc031) values, from which you can get `Device` objects that are used to manage vehicles. 
+
+The following code fragment shows how to iterate through the UUIDs (in this case, just printing them to standard `cout`).
+
+```cpp
+//Iterate through detected devices
+std::vector<uint64_t> device_vector= dc.device_uuids();
+for ( auto i = device_vector.begin(); i != device_vector.end(); i++ ) {
+    std::cout << *i << std::endl;
+}
+```
+
+> **Note** The vector contains UUIDs for all devices detected by DroneCore, including those that may no longer be connected. 
+Since at time of writing there is no way to query a device to determine if it is connected, this mechanism is not particularly useful.
+
+
+## Accessing Devices
+
+Once `DroneCore` has provided you with a vehicle `UUID` you can use the [DroneCore::device()](../api_reference/classdronecore_1_1_drone_core.md#classdronecore_1_1_drone_core_1a5bac6e419e56a1f77a51adef98e94e7c) method to get its associated [Device](../api_reference/classdronecore_1_1_device.md) object:
+
+```cpp
+DroneCore dc;
+//... 
+//Use UUID named uuid to get the associated Device
+Device &device = dc.device(uuid);
+```
+
+> **Tip** If you know that there is only one connected device you can instead call `device()` without any UUID. This will 
+return the first device detected, or a null `Device` if none have been discovered.

```

Comment:
@julianoes 
- What are the properties of a Null device - ie can you access an Action for it? etc.
- Really, all I am after here is to understand  what is the easiest way to tell that you have a null device

Reply:
@julianoes Thanks. Merged. Can you tell me the best way to check whether a Device from device() is valid - ie is Null?